packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1286,13 +1271,9 @@ export class TransactionController extends BaseController<
     return { meta: transaction, isCompleted };
   }
 
-  private getChainAndNetworkId(): {
-    networkId: string | null;
-    chainId: Hex;
-  } {
-    const { networkId, providerConfig } = this.getNetworkState();
-    const chainId = providerConfig?.chainId;
-    return { networkId, chainId };
+  private getChainId(): Hex {
+    const { providerConfig } = this.getNetworkState();
+    return providerConfig?.chainId;
```

Comment:
Is the `?.` necessary? It seems that this function always returns `Hex` and not `Hex | undefined`.